### ISSUE TITLE
VFST replace Jenkins with Github Actions

### ISSUE THREAD

**Issue by oseasmoran73** [department-of-veterans-affairs/va.gov-team#33419]
## Description

As the work for `content-build` and `vets-website` has been recently completed, continuing with this initiative, we would like to deprecate Jenkins in VFST.

**Comment by oseasmoran73** [comment]
## Plan

We want to follow a similar plan/structure that we did for `vets-website` and `content-build`. For full transparency, it most likely will not linger as long as the other two repos did given that we have metrics on runners and workflows.

Given that there are only 4 steps (lint, build, test and deploy [master]), I propose the following:

1. Merge PR
2. We will want at least a week to monitor activity and see if the results of Github Actions match Jenkins
3. Notify team via Slack of the transition in VFST & GHA merge blocker. Given activity, we can probably give a 2 days worth of notice
4. Make Github Action merge blocker, Jenkins will still run in parallel
5. Notify team via Slack of transition in VFST & deploys for documentation
4. Merge PR to enable deploy in Github Action and disable deploy in Jenkins
  1. Will need to monitor this closely as not able to test deploy of gatsby site
5. Disable Jenkins and remove from CI


@meganhkelley @timwright12 Thoughts?

**Comment by meganhkelley** [comment]
What do you mean by "it most likely will not linger as long"? What is it? 

Also, let's add a step for "generate test plan and complete manual QA" — we need to make sure we've documented all the current supported functionality of the pipeline in Jenkins and that it's supported in GHA. That will be a lot simpler than it was in vets-website and content-build, but we still need to document it. 

Thanks! This looks great!

**Comment by oseasmoran73** [comment]
> it most likely will not linger as long

I was referring that `vets-website` and `content-build` took a few months to fully be out the door. I don't suspect that will be the case for VFST.

Will update section with test plan and complete manual QA!

**Comment by oseasmoran73** [comment]
I synced up with DST regarding migrating to Github Actions. The only thing `deploy` step does it update the gatsby site. [Formation](https://department-of-veterans-affairs.github.io/veteran-facing-services-tools/getting-started/common-tasks/updating-formation/) deploys are done manually.

## Test Plan

Currently, CI is comparing Github Actions to Jenkins. All the implementation is done, with the exception of `deploy` which runs against `master` only 

[Deploy](https://department-of-veterans-affairs.github.io/veteran-facing-services-tools/getting-started/common-tasks/updating-formation/) is done on merge of the PR. When we do the migration, we can test a successful `deploy` by creating a separate PR to update a text in the gatsby site. Upon success, we can verify that `deploy` is working as intended similarly to how Jenkins was

## QA

We can only QA `deploy` step. As outlined above, the easiest way to test is upon enabling `deploy` in Github Actions, create a PR to update something in gatsby site. Upon confirmation that it is updating accordingly


@meganhkelley @timwright12 Thoughts?

**Comment by timwright12** [comment]
@oseasmoran73 I think that works

**Comment by oseasmoran73** [comment]
@timwright12 There is one small action pending for this ticket. I am blocked due to reviewing [Jenkins file](https://github.com/department-of-veterans-affairs/veteran-facing-services-tools/blob/master/Jenkinsfile#L53-L54) requiring username/password to deploy. We only have `TOKEN` and not `PASSWORD` (to my knowledge). We would need to request access to be able to retrieve via ssm

It also may make sense to have username in ssm as well. While we can hard-code it, does not seem best practice. Thoughts?

**Comment by timwright12** [comment]
This seems like it might be complete, @mydesignrocks can we remove Jenkins entirely at this point from VFST?

**Comment by mydesignrocks** [comment]
yes the deploy is all moved, we can do the jenkins too

**Comment by theodur** [comment]
@mydesignrocks I think the followup items to remove Jenkins entirely are:
- [ ] Remove the [Jenkinsfile](https://github.com/department-of-veterans-affairs/veteran-facing-services-tools/blob/master/Jenkinsfile) and [credential-helper](https://github.com/department-of-veterans-affairs/veteran-facing-services-tools/blob/master/scripts/credential-helper.sh)
- [ ] Remove the Jenkins [webhook](https://github.com/department-of-veterans-affairs/veteran-facing-services-tools/settings/hooks) for the repo
- [ ] Remove `veteran-facing-services-tools` from the [seed job](https://github.com/department-of-veterans-affairs/devops/blob/master/ansible/deployment/config/jenkins-vetsgov/seed_job.groovy#L1176-L1179)
- [ ] Manually remove the [job](http://jenkins.vfs.va.gov/job/testing/job/veteran-facing-services-tools/) from Jenkins

I might be missing something

**Comment by timwright12** [comment]
@mydesignrocks is this complete?

**Comment by mydesignrocks** [comment]
cleanup is yet to be done, let me knock it off today @timwright12

**Comment by mydesignrocks** [comment]
doesnt look like I have access to remove a job from within jenkins and follow up with infra team but this should ensure that the jobs doesnt get triggered from jenkins.

**Comment by oseasmoran73** [comment]
@mydesignrocks Merged PR on your behalf and updated `seed_job` in Jenkins
